### ISSUE TITLE
REST API missing endpoint for sourcestamps

### ISSUE THREAD

**Issue by aakashja** [buildbot/buildbot#3458]
In buildbot 0.8, rest api for step contains information about sourcestamps (e.g.: https://buildbot.buildbot.net/json/builders/db-postgres/builds/746 contains sourcestamps) 

However, in 0.9 neither step information contain sourcestamps information, nor there is a corresponding api "/builders/n:builderid/builds/n:build_number/sourcestamps/".  To make things worst, the new api "/sourcestamps/n:ssid" requires ssid, however none of the build information from rest api conains ssid (e.g.: https://nine.buildbot.net/api/v2/builders/2/builds/126647 doesn't have ssid).

Use-case: We have a custom dashboard (currently using buildbot 0.8) which extract sourcestamp information from build step information. However, there doesn't seems to any efficient way in 0.9 to fetch sourcestamp for a given build.

I suggest that following endpoint should be added to the rest api: `/builders/n:builderid/builds/n:build_number/sourcestamps/`

**Comment by tardyp** [comment]
Usually. you dont want to extract sourcestamps per build in a UI dashboard. This would make one REST api per build and kill perforamce.
What we do is to fetch list of builds list of recent buildsets, list of recent change, and match them in javascript.
You can see the console_view or grid_view source code for details.

Feel free to add the data API if you think it is really needed

**Comment by aakashja** [comment]
I agree that the approach you described might be more efficient while fetching a large number of sourcestamps.

I still think that this endpoint should exist in REST API for two reasons:

1) while designing the system (REST API), it should have the flexibility for the user to choose. The user might want to fetch just few sourcestamps (only of specific failing builds), in that case fetching all the sourcestamps (which might be in thousands) would be inefficient.

2) compatibility with legacy systems (0.8-regression). Any system using buildbot 0.8 and fetching build data used to get sourcestamp info. Now there is no way for them to get the sourcestamp info with the build info. Build info doesn't even have ssid (using which we can fetched sourcestamp).

I probably can make my dashboard work without this information, however it might be good addition to buildbot REST API.
